homeassistant/components/sensor/__init__.py

```diff
@@ -501,7 +501,7 @@ def state(self) -> Any:
                     f"Sensor {self.entity_id} is providing enum options, but {reason}"
                 )
 
-            if (options := self.options) and value not in options:
+            if (options := self.options) and str(value) not in options:
```

Comment:
The `enum` device class is used for sensors representing non-numeric values (strings).
This should be handled in the integration itself, not here.